.github/workflows/create-release-proposal.yml

```
@@ -0,0 +1,75 @@
+# This action requires the following secrets to be set on the repository:
+#   GH_USER_NAME: GitHub user whose Jenkins and GitHub token are defined below
+#   GH_USER_TOKEN: GitHub user token, to be used by ncu and to push changes
+#   JENKINS_TOKEN: Jenkins token, to be used to check CI status
+
+name: Create Release Proposal
+
+on:
+  workflow_dispatch:
+    inputs:
+      release-line:
+        required: true
+        type: number
+        default: 23
+        description: 'The release line (without dots or prefix). e.g: 22/20/18'
+      release-date:
+        required: true
+        type: string
+        default: YYYY-MM-DD
+        description: The release date in YYYY-MM-DD format
+
+concurrency: ${{ github.workflow }}
+
+env:
+  NODE_VERSION: lts/*
+
+permissions:
+  contents: read
+
+jobs:
+  env:
+  releasePrepare:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332  # v4.1.7
+        with:
+          branch: ${{ env.STAGING_BRANCH }}
+          # Needs the whole git history for ncu to work
+          # See https://github.com/nodejs/node-core-utils/pull/486
+          fetch-depth: 0
+
+      # Install dependencies
+      - name: Install Node.js
+        uses: actions/setup-node@1e60f620b9541d16bece96c5465dc8ee9832be0b  # v4.0.3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Install @node-core/utils
+        run: npm install -g @node-core/utils
+
+      - name: Set variables
+        run: |
+          echo "REPOSITORY=$(echo ${{ github.repository }} | cut -d/ -f2)" >> $GITHUB_ENV
+          echo "OWNER=${{ github.repository_owner }}" >> $GITHUB_ENV
+          echo "STAGING_BRANCH=v${{ inputs.release-line }}.x-staging" >> $GITHUB_ENV
+          echo "RELEASE_BRANCH=v${{ inputs.release-line }}.x" >> $GITHUB_ENV
+          echo "RELEASE_DATE=${{ inputs.release-date }}" >> $GITHUB_ENV
```

I have a few concerns about this:
1. IMO as many variables as possible should be set outside of the bash environment
2. For the variables that must be set in the bash environment, they should be set in a way that doesn't have any expressions inside the bash environment:

```suggestion
          echo "REPOSITORY=$(echo $GITHUB_REPOSITORY  | cut -d/ -f2)" >> $GITHUB_ENV
```